### PR TITLE
Remove leak error in HomeScreenFragment during Settings Clear

### LINKED ISSUE #274: Settings Clear Dialog Causing Window Leak error
What is wrong?
The Dialog to clear the cache is causing a Window Leak Error

Steps to reproduce the issue.

Go to HomeScreenFragment, press on the settings button, then yes. Follow the logcat.

Expected behaviour :
Error is probably being caused becuse Dialog lifetime is exceeding Fragment Lifetime, which shouldn't have happened.

Development environment
1. Device model - Nexus-5 (Emulator)
2. Android version - API 24

LogCat :

 11-14 00:42:38.392 2417-2417/com.peacecorps.malaria E/WindowManager: android.view.WindowLeaked: Activity com.peacecorps.malaria.activities.MainActivity has leaked window DecorView@c6b7ffd[Reset Data] that was originally added here
                                                                         at android.view.ViewRootImpl.<init>(ViewRootImpl.java:417)
                                                                         at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:331)
                                                                         at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
                                                                         at android.app.Dialog.show(Dialog.java:316)
                                                                         at com.peacecorps.malaria.fragment.HomeScreenFragment.addDialog(HomeScreenFragment.java:529)
                                                                         at com.peacecorps.malaria.fragment.HomeScreenFragment$1.onClick(HomeScreenFragment.java:102)
                                                                         at android.view.View.performClick(View.java:5610)
                                                                         at android.view.View$PerformClick.run(View.java:22265)
                                                                         at android.os.Handler.handleCallback(Handler.java:751)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                         at android.os.Looper.loop(Looper.java:154)
                                                                         at 
android.app.ActivityThread.main(ActivityThread.java:6077)
                                                                         at java.lang.reflect.Method.invoke(Native Method)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)

Do I want to work on the issue?
If assigned, then yes.

@anubhakushwaha  Please review this issue once

### PR DESCRIPTION
**Pull request by DipanshKhandelwal**
### Description
Remove leak error in HomeScreenFragment in Settings Clear Dialog

Fixes #274 

### Type of Change:
- Code: Dismiss dialog in onStop method.

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: malaria-app-android/src/main/java/com/peacecorps/malaria/fragment/HomeScreenFragment.java [hunk 1]
```diff
@@ -473,6 +473,16 @@ public void missedDayRecord(int day, int month, int year){
 
     }
 
+    @Override
+    public void onStop() {
```
Comment (m-murad) [inline malaria-app-android/src/main/java/com/peacecorps/malaria/fragment/HomeScreenFragment.java]: There is no guarantee that the `onStop` method will be called when we leave an activity. You need to move the code to `onPause` method.
For more info please read - https://developer.android.com/reference/android/app/Activity.html

**Review by m-murad — CHANGES_REQUESTED** [review]

File: malaria-app-android/src/main/java/com/peacecorps/malaria/fragment/HomeScreenFragment.java (hunk 1 above)
Comment (DipanshKhandelwal) [reply malaria-app-android/src/main/java/com/peacecorps/malaria/fragment/HomeScreenFragment.java]: @m-murad Oh sorry. I'll do it.

**Comment by DipanshKhandelwal** [comment]
@m-murad Please take a look.

**Comment by mayburgos** [comment]
please send a new pr with the updated code from master. thank you!

**Pull request closed without merging**